src/post/Write/Write.js

```diff
@@ -136,33 +140,32 @@ export const CREATE_POST_ERROR = 'CREATE_POST_ERROR';
 
 const requiredFields =
   'parentAuthor,parentPermlink,author,permlink,title,body,jsonMetadata'
-  .split(',');
-function createPost(body) {
+    .split(',');
+function createPost(postData) {
   requiredFields.forEach((field) => {
     assert(
-      body[field] != null,
+      postData[field] != null,
       `Developer Error: Missing required field ${field}`
     );
   });
-
+  const { parentAuthor, parentPermlink, author, permlink, title, body, jsonMetadata } = postData;
   return () => ({
     type: CREATE_POST,
     payload: {
-      promise: request.get(`${process.env.STEEMCONNECT_HOST}/comment`)
-        .query(body)
-        .withCredentials()
-        .endAsync()
-        .then((res) => res.body)
-        .tap(() => {
-          browserHistory.push(
-            `/${body.parentPermlink}/@${body.author}/${body.permlink}`
-          );
-        }),
+      promise: SteemConnect
+        .commentAsync(parentAuthor, parentPermlink, author, permlink, title, body, jsonMetadata)
+        .then((result) => {
+          if (result !== null) {
+            throw new Error(result);
+          } else {
+            browserHistory.push(`/${parentPermlink}/@${author}/${permlink}`);
```

Comment:
The redirection is not working actually. There should be a loading after submit and a redirection when its been confirmed.